linuxcomport.cpp

```diff
@@ -7,91 +7,98 @@
 
 #include "linuxcomport.h"
 
-void WaitForAnswer(int fileDescriptor) {
-	fd_set rfds;
-	FD_ZERO(&rfds);
-	FD_SET(0, &rfds);
-	struct timeval timeout;
-	timeout.tv_sec = 0;
-	timeout.tv_usec = 50000;
-	select(fileDescriptor, &rfds, NULL, NULL, &timeout);
+void WaitForAnswer(int fileDescriptor)
+{
+    fd_set rfds;
+    FD_ZERO(&rfds);
+    FD_SET(0, &rfds);
+    struct timeval timeout;
+    timeout.tv_sec = 0;
+    timeout.tv_usec = 50000;
+    select(fileDescriptor, &rfds, NULL, NULL, &timeout);
 }
 
-size_t linuxComPort::LinuxComPort::sendData(const char* buffer, const size_t length) {
-	size_t count = write(fileDescriptor, buffer, length);
-	return count == length;
+size_t linuxComPort::LinuxComPort::sendData(const char* buffer, const size_t length)
+{
+    size_t count = write(fileDescriptor, buffer, length);
+    return count == length;
```

Comment:
So, hier ist ein super Beispiel fuer das Problem mit deinen Rueckgabewerten. Was wird die Funktion zurueckgeben, wenn alles gepasst hat? --> 1 , weil das true vom Compiler zu einen size_t gecastet wird. Jetzt ist die Frage, erwartet das der Anwender deiner Funktion. Eine Funktion mit diesem Aussehen erinnert mich immer an Socket Funktionen: http://man7.org/linux/man-pages/man2/write.2.html Hier wird der Return-Type jedoch komplett anders verwendet. Was noch schlimmer ist, durch die definition deiner Return Types in der communication.h Datei, erwarte ich 0, wenn alles geklappt hat. Allein aus dem Betrachten der Header Dateien werde ich mir das Verhalten deine Funktionen so nicht erklaeren koennen. Daher, Entscheide dich fuer eine Strategie, und setz diese Konsequent um.